texar/tf/utils/dynamic_decode.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utility functions for decoding.
```

Comment:
This is adapted from TF. So, let's keep TF's original copyright header (not Texar's), and add a line indicating the modification. Refer to: https://github.com/asyml/texar/blob/master/texar/tf/modules/decoders/tf_helpers.py#L15

Besides, I think it's better to put this file under `modules/decoders`

Reply:
In the file's docstring, briefly add a line indicating the modification. E.g., "This file is modified from tf.contrib.seq2seq.dynamic_decode"